### PR TITLE
Add support for VK_EXT_private_data extension.

### PR DESCRIPTION
**Pull request by billhollings**
Moved to a new model for creation: create and potentially destroy the object
within `MVKDevice::create...`, to hide it from `vulkan.mm` where all other object
creation errors are handled. We could move to this slowly over time.

Passes all 49 private data CTS tests.

Implements issue #877.

### CODE REVIEW

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.h [hunk 1]
```diff
@@ -678,6 +686,7 @@ class MVKDevice : public MVKDispatchableVulkanAPIObject {
 	const VkPhysicalDeviceScalarBlockLayoutFeaturesEXT _enabledScalarLayoutFeatures;
 	const VkPhysicalDeviceTexelBufferAlignmentFeaturesEXT _enabledTexelBuffAlignFeatures;
 	const VkPhysicalDeviceVertexAttributeDivisorFeaturesEXT _enabledVtxAttrDivFeatures;
+	const VkPhysicalDevicePrivateDataFeaturesEXT _enabledPrivateDataFeatures;
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:689]: I like to keep these sorted like they are in MVKExtensions.def.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 2]
```diff
@@ -169,6 +169,11 @@
 				divisorFeatures->vertexAttributeInstanceRateZeroDivisor = true;
 				break;
 			}
+			case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PRIVATE_DATA_FEATURES_EXT: {
+				auto* privateDataFeatures = (VkPhysicalDevicePrivateDataFeaturesEXT*)next;
+				privateDataFeatures->privateData = true;
+				break;
+			}
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:176]: Same here.

File: MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm [hunk 3]
```diff
@@ -669,6 +669,10 @@
 	ADD_DVC_EXT_ENTRY_POINT(vkCmdDebugMarkerInsertEXT, EXT_DEBUG_MARKER);
 	ADD_DVC_EXT_ENTRY_POINT(vkGetRefreshCycleDurationGOOGLE, GOOGLE_DISPLAY_TIMING);
 	ADD_DVC_EXT_ENTRY_POINT(vkGetPastPresentationTimingGOOGLE, GOOGLE_DISPLAY_TIMING);
+	ADD_DVC_EXT_ENTRY_POINT(vkCreatePrivateDataSlotEXT, EXT_PRIVATE_DATA);
+	ADD_DVC_EXT_ENTRY_POINT(vkDestroyPrivateDataSlotEXT, EXT_PRIVATE_DATA);
+	ADD_DVC_EXT_ENTRY_POINT(vkGetPrivateDataEXT, EXT_PRIVATE_DATA);
+	ADD_DVC_EXT_ENTRY_POINT(vkSetPrivateDataEXT, EXT_PRIVATE_DATA);
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm:675]: And here as well.

File: MoltenVK/MoltenVK/Vulkan/vulkan.mm [hunk 4]
```diff
@@ -2946,13 +2947,68 @@ MVK_PUBLIC_SYMBOL VkResult vkGetPastPresentationTimingGOOGLE(
...
 	MVKSwapchain* mvkSwapchain = (MVKSwapchain*)swapchain;
 	VkResult rslt = mvkSwapchain->getPastPresentationTiming(pPresentationTimingCount, pPresentationTimings);
 	MVKTraceVulkanCallEnd();
 	return rslt;
 }
 
+#pragma mark -
+#pragma mark VK_EXT_private_data extension
+
+MVK_PUBLIC_SYMBOL VkResult vkCreatePrivateDataSlotEXT(
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/Vulkan/vulkan.mm:2961]: And these functions as well.

**Review by cdavis5e — CHANGES_REQUESTED** [review]
Looks good. Just a minor note on extension ordering.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.h (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.h]: Done.

File: MoltenVK/MoltenVK/Vulkan/vulkan.mm (hunk 4 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/Vulkan/vulkan.mm:2933]: Done.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 2 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: Done.

File: MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm (hunk 3 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm:673]: Done.

**Comment by billhollings** [comment]
Huh...not an easy task mapping calls/structs to extensions...since not all extensions have either, and when they do their names don't always align simply...but I think I've got ordering right.

I didn't do a wholesale review of the ordering of other extensions and extension components, but I did adjust a couple that I did notice while I was working on these.

**Review by cdavis5e — APPROVED** [review]

**Pull request merged**